src/type-check.rkt

```diff
@@ -34,31 +41,47 @@
   (unless (null? errs)
     (raise-herbie-syntax-error "Program has type errors" #:locations errs)))
 
+(define (type-of expr env)
+  (expression->type (datum->syntax #f expr) env
+                    (lambda (stx msg . args)
+                      (error "Unexpected call to error! within type-of"
+                             stx (apply format msg args)))))
+
+
 (define (expression->type stx env error!)
   (match stx
     [(or #`TRUE #`FALSE) 'bool]
     [#`,(? constant? x) 'real]
     [#`,(? variable? x) (dict-ref env x)]
     [#`(,(and (or '+ '- '* '/) op) #,exprs ...)
+     (define t #f)
```

Comment:
I think this is almost correct, but it fails for `(+)`, which actually is valid and returns `0`. I think `t` should be initialized to `'real`, and also you should probably call the variable something else.

Reply:
fixed and added a test for (+)